### ISSUE TITLE
Change graphql tag after build

### ISSUE THREAD

**Issue by Faliszek** [reasonml-community/reason-apollo-client#70]
Hi, first of all thanks for awesome library, really enjoy it by far.

I have question is there a way to modify output of graphql query?
I really want to remove any `require` statements from my .bs.js file.
Right now my query look like this:

```
/* Me_Query.re */
[%graphql {|
  query Me {
      me {
        firstName
        id
        lastName
        ...
      }
  }
|}];
```

and builded file like this

```
/* Me_Query.bs.js */

import * as ApolloClient__React_Hooks_UseQuery from "reason-apollo-client/src/@apollo/client/react/hooks/ApolloClient__React_Hooks_UseQuery.bs.js";

var Raw = {};

var query = (require("@apollo/client").gql`
  query Me  {
    me  {
      __typename
      firstName
      id
      lastName
    }
  }
`)

```

Is there any way I can from 
**require("@apollo/client").gql**

change it to something like this:

**import {gql} from 'graphql-tag'**
**var query = gql`...`**

Thanks in advance.

**Comment by jfrolich** [comment]
That is the only way to do the template literal (as far as I know) with ReScript. To do your version we need a change in the ReScript compiler to allow for this, I discussed this with Hongbo multiple times, and he said that it's going to land at some point. Don't hold your breath because this is a long-standing issue (almost a year).

See:

https://github.com/rescript-lang/rescript-compiler/issues/4294
https://github.com/rescript-lang/rescript-compiler/issues/4070

**Comment by Faliszek** [comment]
@jfrolich thanks for quick response!
Oh cool, glad to hear.

So if by any chance someone else want to use snowpack and reason-apollo-client with graphql-ppx and encounter "Uncaught error" caused by 'require' in .bs.js file. 

Here is what I did - but use it at your own risk we violate lot's of best practices here 😅

In my project I store mutations and queries in folders
```
/src/mutations
/src/queries
```
And files with mutations and queries are just `[%graphql {| ... |}];` without any additional logic in it.

1. Add graphql-tag `yarn add graphql-tag` - right now is the newest is 2.11.0
2. Add to bsconfig js-post-build (it will execute after compiling every single .bs.js - so this is not good, but works)
```
//bsconfig.json
 "package-specs": [
    {
      "module": "es6",
      "in-source": true
    }
  ],
  "graphql": {
    "apolloMode": true,
    "extendMutation": "ApolloClient.GraphQL_PPX.ExtendMutation",
    "extendQuery": "ApolloClient.GraphQL_PPX.ExtendQuery",
    "extendSubscription": "ApolloClient.GraphQL_PPX.ExtendSubscription",
    "templateTagReturnType": "ApolloClient.GraphQL_PPX.templateTagReturnType",
    "templateTagImport": "gql",
    "templateTagLocation": "@apollo/client"
  },
  "js-post-build": {
    "cmd": "./graphql-tag.sh"
  },
```
3. Create file in /src called `graphql-tag.sh`
https://gist.github.com/Faliszek/d66449c8a6837a013e59143b5f92df0f

Right now I have like 5 queries and 5 mutations and i didn't notice any slowing down in build.
It is risky but it is only to developing and having super fast rebuild in browser with snowpack ;p

**Comment by jfrolich** [comment]
Yes the problem is that you cannot have multiple queries in one file and no code before the query. Probably it can also be solved using a babel plugin.

Another solution is to not use the template tag, but just a plain string for the query. That worked before. I am not sure if `reason-apollo-client` requires the template tag?

**Comment by jfrolich** [comment]
Perhaps I can add configuration to call a function on the query instead of the template tag (`gql` can also be called as a function). That will definitely work. You lose some performance benefits etc. of the possible `gql` babel plugin if you don't use the template tag.

**Comment by jeddeloh** [comment]
> Another solution is to not use the template tag, but just a plain string for the query (remove all template tag configuration). That worked before. I am not sure if reason-apollo-client requires the template tag? (I guess it requires it because otherwise it needs to wrap the string query in a gql function at runtime).

We do require the template tag right now. Perhaps I could do some magic and inspect what's coming in as query and dynamically run through `gql` if necessary.

> Perhaps I can add configuration to call a function on the query instead applying a template tag (gql can also be called as a function). That will definitely work. You lose some performance benefits etc. of the possible gql babel plugin if you don't use the template tag.

Yes, that would probably be better than my proposal

**Comment by jfrolich** [comment]
Ok I created a configuration option:

See: 
https://github.com/reasonml-community/graphql-ppx/commit/5d92a6bd8802049141511c809b75087ecd121823

You can add the `"templateTagIsFunction": true` to your `bsconfig.json` or `ppxConfig` in a query.

It will be automatically pushed to npm under `1.0.2-ca56c96.0`

**Comment by Faliszek** [comment]
Thanks, i will give it a shot in couple days, see if it works, awesome work

**Comment by Faliszek** [comment]
@jfrolich sorry for late response.

Your fix works perfectly.

So here is what i did
1. Remove hacky solution from bsconfig.json
2. Update graphql-ppx to 1.0.2-ca56c96.0
3 .Add to bsconfig.json, "graphql": { "templateTagIsFunction": true}
and after `bsb -clean-world` and `bsb -make-world -w`

I get result 

```
import * as Client from "@apollo/client";
import * as ApolloClient__React_Hooks_UseMutation from "reason-apollo-client/src/@apollo/client/react/hooks/ApolloClient__React_Hooks_UseMutation.bs.js";

var Raw = {};

var query = Client.gql("mutation UpdateClient($clientId: Int!,  ...");
```

So i would say, it works perfectly! Thanks for your help.

**Comment by jfrolich** [comment]
@Faliszek: It has been released as 1.0.2, see changelog here: https://graphql-ppx.com/docs/changelog
